shubhrajps/stage-stories
# [MAJOR] Added Stories Feature for STAGE

# Added stories feature for STAGE
Deployed at https://stage-stories.vercel.app/

## Changes Made

1. New Components:
- UserStories: The main component that handles the display of user stories based on the selected user.
- StoryViewer: Displays the selected user's stories in a viewer format.
- UserCircle: Represents each user in a circular format, allowing users to select a user to view their stories.
- UserCircleList: A list of UserCircle components for all users.

2. Context Management:
- StoryContext: A context to manage the state of selected users and all user stories, making it accessible throughout the application.

3. API Route:
- fetchAllStories: An API route that returns all user stories as a JSON response.

4. Responsive Design:
- Implemented responsive behavior in the UserStories component to switch between mobile and desktop views based on the window size.

5. Testing:
- Added unit tests for all components using Jest and React Testing Library.
- Created integration tests for the API route to ensure it returns the correct data.

6. CI/CD Pipeline:
- Set up a GitHub Actions workflow to run tests and deploy to Vercel upon pushing changes to the main branch or creating a pull request.

## Screenshots & Videos

### Homescreen
<img width="376" alt="Screenshot 2024-09-01 at 12 29 55 AM" src="https://github.com/user-attachments/assets/ebf51203-823f-4160-b3e9-de7c3528e116">

### StoriesView
<img width="379" alt="Screenshot 2024-09-01 at 12 30 01 AM" src="https://github.com/user-attachments/assets/dfe54ada-e315-44ae-b840-6191b45b5d41">

### Live Video
https://github.com/user-attachments/assets/f577bbdf-6ceb-4b80-b148-ab196ab14043

## How to test
1. Clone the repository `git clone https://github.com/shubhrajps/stage-stories.git`
2. Check the branch `git checkout add-stories-feature`
3. Install the dependencies `npm install`
4. Run the project `npm run dev`
5. Access the project at `https://localhost:3000`

## Checklist
- [ ]  Tests have been written and pass successfully.
- [ ]  CI/CD pipeline is configured and working.
- [ ]  Changes have been tested in both mobile and desktop views.
